.github/workflows/docker-build-and-run.yml

```diff
@@ -0,0 +1,85 @@
+name: Build and Run Docker
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    types: [opened, synchronize, edited]
+    branches:
+      - main
+      - "docker/**"
+jobs:
+  build-and-run-docker:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.x'
+
+      - name: Install Python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+
+      - name: Generate input files
+        run: |
+          python -m Docker.generate_signal_docker_test
+
+      - name: Verify input files
+        run: |
+          for file in ivim_simulation.nii.gz ivim_simulation.bval ivim_simulation.bvec; do
+            if [ ! -f "$file" ]; then
+              echo "Error: $file not found"
+              exit 1
+            fi
+          done
+          echo "All input files generated successfully"
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+
+      - name: Build Docker image
```

Comment:
I think docker buildx can build directly to files `docker buildx build --output type=tar,dest=my-image.tar .`

Reply:
Thanks for this.

Reply:
Yeah, it works, but, it directly save the docker image into the .tar file and we have some following line which uses the docker image and needs to be loaded again. I felt it is more cleaner having it like this.